store/sqlstore/plugin_store.go

```diff
@@ -219,16 +220,39 @@ func (ps SqlPluginStore) SetWithOptions(pluginId string, key string, value []byt
 }
 
 func (ps SqlPluginStore) Get(pluginId, key string) (*model.PluginKeyValue, *model.AppError) {
-	var kv *model.PluginKeyValue
 	currentTime := model.GetMillis()
-	if err := ps.GetReplica().SelectOne(&kv, "SELECT * FROM PluginKeyValueStore WHERE PluginId = :PluginId AND PKey = :Key AND (ExpireAt = 0 OR ExpireAt > :CurrentTime)", map[string]interface{}{"PluginId": pluginId, "Key": key, "CurrentTime": currentTime}); err != nil {
+
+	failure := func(err error, statusCode int) (*model.PluginKeyValue, *model.AppError) {
+		return nil, model.NewAppError(
+			"SqlPluginStore.Get",
+			"store.sql_plugin_store.get.app_error",
+			nil,
+			fmt.Sprintf("plugin_id=%v, key=%v, err=%v", pluginId, key, err.Error()),
+			statusCode,
+		)
+	}
+
+	query := ps.getQueryBuilder().Select("PluginId, PKey, PValue, ExpireAt").
+		From("PluginKeyValueStore").
+		Where(sq.Eq{"PluginId": pluginId}).
+		Where(sq.Eq{"PKey": key}).
+		Where(sq.Or{sq.Eq{"ExpireAt": 0}, sq.Gt{"ExpireAt": currentTime}})
+	queryString, args, err := query.ToSql()
+
+	if err != nil {
+		return failure(err, http.StatusInternalServerError)
+	}
+	row := ps.GetReplica().Db.QueryRow(queryString, args...)
+
+	var kv model.PluginKeyValue
+	if err := row.Scan(&kv.PluginId, &kv.Key, &kv.Value, &kv.ExpireAt); err != nil {
 		if err == sql.ErrNoRows {
-			return nil, model.NewAppError("SqlPluginStore.Get", "store.sql_plugin_store.get.app_error", nil, fmt.Sprintf("plugin_id=%v, key=%v, err=%v", pluginId, key, err.Error()), http.StatusNotFound)
+			return failure(err, http.StatusNotFound)
```

Comment:
nit: it feels slightly more obvious to write:
```
return nil, failure(err, http.StatusNotFound)
```

instead of having `failure` always return a `nil` `*model.PluginKeyValue`. 0/5